x/emissions/keeper/query_server.go

```diff
@@ -226,3 +226,54 @@ func (qs queryServer) GetRegisteredTopicsIds(ctx context.Context, req *state.Que
 
 	return &state.QueryRegisteredTopicsIdsResponse{TopicsIds: topicsIds}, nil
 }
+
+func (qs queryServer) GetInferenceRequest(ctx context.Context, req *state.QueryInferenceRequestRequest) (*state.QueryInferenceRequestResponse, error) {
+	valid := state.IsValidRequestId(req.RequestId)
```

Comment:
surprised we need validation here... guess it doesn't hurt.

Reply:
we don't actually, we could just make the request and then do an if statement on collections.IsNotFound but I think this is more defense in depthable code and it's probably about the same for performance